### PR TITLE
Added a new common control attribute 'system_log_level'

### PR DESCRIPTION
**Pull request by vpodzime**
For specifying the minimum log level required for log messages to
go to the system log.

Ticket: ENT-7594
Changelog: Commit

### CODE REVIEW

**Comment by vpodzime** [comment]
Requires https://github.com/cfengine/libntech/pull/153 to be merged first.

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7143/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7143/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7143/

**Comment by vpodzime** [comment]
@cf-bottom one more round in jenkins, please!

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/7144/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/7144/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-7144/

**Comment by vpodzime** [comment]
> Alright, I triggered a build:
> 
> [Build Status](https://ci.cfengine.com//job/pr-pipeline/7143/)
> 
> ****

Just bad luck.

File: libpromises/mod_common.c [hunk 1]
```diff
@@ -259,6 +259,7 @@ const ConstraintSyntax CFG_CONTROLBODY[COMMON_CONTROL_MAX + 1] =
     ConstraintSyntaxNewContextList("site_classes", "A list of classes that will represent geographical site locations for hosts. These should be defined elsewhere in the configuration in a classes promise.", SYNTAX_STATUS_NORMAL),
     ConstraintSyntaxNewString("syslog_host", CF_IPRANGE, "The name or address of a host to which syslog messages should be sent directly by UDP. Default value: localhost", SYNTAX_STATUS_NORMAL),
     ConstraintSyntaxNewInt("syslog_port", CF_VALRANGE, "The port number of a UDP syslog service. Default value: 514", SYNTAX_STATUS_NORMAL),
+    ConstraintSyntaxNewString("system_log_level", ".*", "Minimum system log level of messages that should be logged to system log", SYNTAX_STATUS_NORMAL),
```
Comment (nickanderson) [inline libpromises/mod_common.c:262]: Definately I think we need some more detailed description of this in the docs for what specific strings users should expect to use.

In the windows event log I see: 
- Critical
- Error
- Warning
- Information
- Audit Success
- Audit Failure

What strings would I use on Linux for syslog?
- Emergency
- Alert
- Critical
- Error
- Warning
- Informational
- Debug
?

Also, I guess the string match is case insensitive, is that correct?

**Review by nickanderson — APPROVED** [review]
I tested this manually on windows.

I can see that after introducing the new attribute event logs from CFEngine seemed to stop.

File: libpromises/mod_common.c (hunk 1 above)
Comment (vpodzime) [reply libpromises/mod_common.c:262]: It needs to be a string that our code recognizes as a valid log level so the same as the `--log-level` option of our binaries.

Comment (vpodzime) [reply libpromises/mod_common.c:262]: I'll change the `.*` to the actual regex with all the log levels.

**Review by olehermanse — APPROVED** [review]
LGTM, thanks :)

**Pull request merged**